waterfall/golang/mux/conn.go

```diff
@@ -0,0 +1,67 @@
+package mux
+
+import (
+	"fmt"
+	"io"
+	"net"
+	"time"
+)
+
+type halfCloser interface {
+	CloseRead() error
+	CloseWrite() error
+}
+
+type rwcConn struct {
+	io.ReadWriteCloser
+}
+
+func (mc *rwcConn) Read(b []byte) (int, error) {
+	return mc.ReadWriteCloser.Read(b)
+}
+
+func (mc *rwcConn) Write(b []byte) (int, error) {
+	return mc.ReadWriteCloser.Write(b)
+}
+
+func (mc *rwcConn) Close() error {
+	return mc.ReadWriteCloser.Close()
+}
+
+func (mc *rwcConn) CloseRead() error {
+	if c, ok := mc.ReadWriteCloser.(halfCloser); ok {
+		fmt.Println("Half Closing: reads ...")
```

Comment:
Is the log spam necessary here and below?

Reply:
removed.